recipes/physx/4.1.1/conanfile.py

```diff
@@ -0,0 +1,310 @@
+import os
+import shutil
+import time
+
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration
+
+physx_license = """Copyright (c) 2019 NVIDIA Corporation. All rights reserved.
+
+Redistribution and use in source and binary forms, with or without
+modification, are permitted provided that the following conditions
+are met:
+ * Redistributions of source code must retain the above copyright
+   notice, this list of conditions and the following disclaimer.
+ * Redistributions in binary form must reproduce the above copyright
+   notice, this list of conditions and the following disclaimer in the
+   documentation and/or other materials provided with the distribution.
+ * Neither the name of NVIDIA CORPORATION nor the names of its
+   contributors may be used to endorse or promote products derived
+   from this software without specific prior written permission.
+
+THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS ``AS IS'' AND ANY
+EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+(INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE."""
+
+class PhysXConan(ConanFile):
+    name = "physx"
+    description = "The NVIDIA PhysX SDK is a scalable multi-platform " \
+                  "physics solution supporting a wide range of devices, " \
+                  "from smartphones to high-end multicore CPUs and GPUs."
+    homepage = "https://github.com/NVIDIAGameWorks/PhysX"
+    topics = ("conan", "PhysX", "physics")
+    url = "https://github.com/conan-io/conan-center-index"
+    license = "BSD-3-Clause"
+    exports_sources = ["CMakeLists.txt", "patches/**"]
+    generators = "cmake"
+    settings = "os", "compiler", "arch", "build_type"
+    options = {
+        "build_type": ["debug", "checked", "profile", "release"],
```

Comment:
Are those "checked" and "profile" build types really used? What do they mean? Could we just drop them?

Reply:
We can't drop them:
"checked" is RelWithDebInfo with several defines. It is the recommended PhysX build_type during development.
"profile" is Release with still several defines providing information to identify performance bottlenecks.
There is no equivalent to MinSizeRel.

PhysX documentation provides Best practices referring to those build_type, therefore I think dropping them would be misleading for developers.

Reply:
ok, so let's remove the release and debug options and just left the custom ones. As those are build types only valid with the "Release" build type setting, let's just check that and raise an invaild configuration if not. We would have to manage the package ID accordingly as well.

Something like this would make it:

- In config_options(): if settings.build_type == Debug, then remove self.options.build_type
- In options: Use `None` value as default (for pure release build type)

Reply:
correction: "checked" is like RelWithDebInfo, but compiled with -O3 on gcc

Honnestly I don't really see how to implement in details what you suggest, and if it really handles all cases.

Why should it raises invalid configuration if build type settings is MinSizeRel and build type option is profile or release?

Reply:
other solution:
- map Debug to debug, RelWithDebInfo to checked, Release to either release or profile depending of an option (which would be deleted if build_type is not Release).
- raise ConanInvalidConfiguraton if MinSizeRel.